_posts/2021-07-01-wildfly-preview-bootable-jar.adoc

```diff
@@ -0,0 +1,207 @@
+                    <layers>
+                        <layer>jaxrs-server</layer>
+                    </layers>
+                    <plugin-options>
+                        <jboss-fork-embedded>true</jboss-fork-embedded>
+                    </plugin-options>
+                </configuration>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>package</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+----
+
+With that setting, the application will now be packaged by default using Bootable Jar when we run `mvn package`.
+
+There are 2 important lines in the code snippet:
+
+[source,xml]
+----
+<feature-pack-location>wildfly-preview@maven(org.jboss.universe:community-universe)</feature-pack-location>
+----
+
+This line means that we will use the `wildfly-preview` feature pack to install and provision the WildFly Server so that it runs with Jakarta EE 9.1. If we had specify instead `wildfly@maven(...)`, we would have install a regular WildFly server to run with Jakarta EE 8.
+
+
+[source,xml]
+----
+<layer>jaxrs-server</layer>
+----
+
+As our application is a simple JAX-RS application, we only provision the `jaxrs-server` layer when we install WildFly in the Bootable Jar. This ensures that we have all we need to run a JAX-RS application and nothing more.
+
+We can now package the application as a Bootable Jar by running the command:
+
+[source,shell]
+----
+$ mvn clean package
+----
+
+Once the Maven command is finished, we can start the application by running the command:
+
+[source,shell]
+----
+$ java -jar target/helloworld-rs-bootable.jar
+----
+
+There are some interesting lines in the log when the application starts:
+
+[source,shell]
+----
+08:36:38,676 INFO  [org.jboss.as] (MSC service thread 1-1) WFLYSRV0049: WildFly Preview 24.0.0.Final (WildFly Core 16.0.0.Final) starting
+----
+
+This line above shows that we are running `WildFly Preview` (and not the regular `WildFly`).
+
+[source,shell]
+----
+08:36:54,301 INFO  [org.jboss.resteasy.resteasy_jaxrs.i18n] (ServerService Thread Pool -- 62) RESTEASY002225: Deploying jakarta.ws.rs.core.Application: class org.jboss.as.quickstarts.rshelloworld.JAXActivator$Proxy$_$$_WeldClientProxy
+----
+
+This line above confirms that we are running with Jakarta EE 9.1 as the application is deployed with `jakarta.ws.rs.core.Application` which is in the namespace for Jakarta EE 9.1.
+
+So we have deployed a Jakarta EE 8 application in a Jakarta EE 9.1 server, let's check that the application works as expected:
+
+[source,shell]
+----
+$ curl http://:8080/rest/json
+
+{"result":"Hello World!"}
+----
+
+== Run with Jakarta EE 9.1 on OpenShift
+
+We are now successfully running our JAX-RS application locally with WildFly EE 9.1 Preview. What do we need to run it on OpenShift?
```

Comment:
It might be clearer to change `WildFly EE 9.1 Preview` to `WildFly (Jakarta EE 9.1) Preview.`

Reply:
Or just `WildFly Preview`

Reply:
that's a fair remark.

Our "preview" feature pack is not named consistently. In wildfly.org it is "try EE 9 Preview", it the downloads pages, it is "WildFly Preview EE 9.1 Distribution" and in the news, it is plain "WildFly Preview".